cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client.go

```diff
@@ -453,15 +469,8 @@ func isReservationNotReady(errorCode, errorMessage string) bool {
 }
 
 func isInvalidReservationError(errorCode, errorMessage string) bool {
-	reservationErrors := []string{
-		"Incompatible AggregateReservation VMFamily",
-		"Could not find the given reservation with the following name",
-		"must use ReservationAffinity of",
-		"The reservation must exist in the same project as the instance",
-		"only compatible with Aggregate Reservations",
-	}
-	for _, rErr := range reservationErrors {
-		if strings.Contains(errorMessage, rErr) {
+	for _, re := range regexReservationErrors {
+		if re.MatchString(errorMessage) {
```

Comment:
I'm concerned that this change will make the matching more restrictive.
Old checkers uses a contains logic, wheres now we're doing a full-match (I think).
This could potentially break existing logic where we would have stacktrace to the errorMessage?

Can we explore if we can do a regex contains?